orchagent/routeorch.cpp

```diff
@@ -1012,13 +1012,16 @@ bool RouteOrch::addNextHopGroup(const NextHopGroupKey &nexthops)
 
     /* Assert each IP address exists in m_syncdNextHops table,
      * and add the corresponding next_hop_id to next_hop_ids. */
+    bool missingNexthop = false;
     for (auto it : next_hop_set)
     {
         if (!m_neighOrch->hasNextHop(it))
         {
-            SWSS_LOG_INFO("Failed to get next hop %s in %s",
+            SWSS_LOG_INFO("Failed to get next hop %s in %s, resolving neighbor",
                     it.to_string().c_str(), nexthops.to_string().c_str());
-            return false;
+            m_neighOrch->resolveNeighborEntry(it, MacAddress());
```

Comment:
@shi-su , as discussed, we wouldn't need to continue retry. This is expected to be resolved only once. Also, lets have a logic to remove such entry from APP_DB once it is resolved. Can we cache such unresolved neighbors and take necessary action?

Reply:
Added a logic to avoid retrying and resolve each neighbor only once.